### PR TITLE
[DNM]Ports tgui from Vorestation for all modules/power-files applicable

### PR DESCRIPTION
**Pull request by TheLordME**
## About The Pull Request
Pulls the tguis for almost all machines in the code/modules/power Folder from vorestation.
Machines affected: APC, Pacman-Generators, TEG, AI supermatter interaction, Gridchecker(doesnt actually have one), Power Net Monitor

## Why It's Good For The Game
Nano_UI is depricated, tgui change has to happen

## Changelog
:cl:
add: Added tgui for APCs.
add: Added tgui for portable Generators
add: Added tgui for TEGs.
add: Added tgui for Powernet monitors.
del: Removed NanoUis where it is replaced.
/:cl:

### CODE REVIEW

**Comment by TheLordME** [comment]
Still requieres a tgui-rebuilt, and proper testing, so far only compileability was tested

File: code/modules/power/apc.dm [hunk 1]
```diff
@@ -767,15 +767,19 @@
...
+/obj/machinery/power/apc/ui_interact(mob/user, datum/tgui/ui = null)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Apc", name) // 510, 460
+		ui.open()
 
+/obj/machinery/power/apc/ui_data(mob/user)
 	var/list/data = list(
 		"locked" = locked,
+		"normallyLocked" = locked,
```
Comment (LetterN) [inline code/modules/power/apc.dm:782]: unused

File: code/modules/power/apc.dm [hunk 2]
```diff
@@ -767,15 +767,19 @@
 		wires.Interact(user)
 		return	//The panel is visibly dark when the wires are exposed, so we shouldn't be able to interact with it.
 
-	return nano_ui_interact(user)
+	return ui_interact(user)
```
Comment (LetterN) [inline code/modules/power/apc.dm:770]: idk if this is still needed as attack hand calls tgui

File: code/modules/power/grid_checker.dm [hunk 3]
```diff
@@ -64,7 +64,7 @@
 	if(opened)
 		wires.Interact(user)
 
-	return nano_ui_interact(user)
+	return ui_interact(user)
```
Comment (LetterN) [inline code/modules/power/grid_checker.dm:67]: same with this

File: code/modules/power/port_gen.dm [hunk 4]
```diff
@@ -294,97 +294,84 @@
 	..()
 	if (!anchored)
 		return
-	nano_ui_interact(user)
+	ui_interact(user)
```
Comment (LetterN) [inline code/modules/power/port_gen.dm:297]: same with this, dunno about attack ai

File: code/modules/power/turbine.dm [hunk 5]
```diff
@@ -347,56 +347,60 @@
...
-		// when the ui is first opened this is the data it will use
-		ui.set_initial_data(data)
-		// open the new ui window
-		ui.open()
-		// auto update every Master Controller tick
-		ui.set_auto_update(TRUE)
+	return data
 
-/obj/machinery/computer/turbine_computer/Topic(href, href_list)
+/obj/machinery/computer/turbine_computer/ui_act(action, list/params, datum/tgui/ui, datum/tgui_state/state)
```
Comment (LetterN) [inline code/modules/power/turbine.dm:379]:
```suggestion
/obj/machinery/computer/turbine_computer/ui_act(action, list/params)
```

File: code/modules/power/sensors/sensor_monitoring.dm [hunk 6]
```diff
@@ -47,12 +47,11 @@
...
 		return
-	nano_ui_interact(user)
-
-// Uses dark magic to operate the NanoUI of this computer.
-/obj/machinery/computer/power_monitor/nano_ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	power_monitor.nano_ui_interact(user, ui_key, ui, force_open)
+	ui_interact(user)
 
+// Uses dark magic to operate the tgUI of this computer.
+/obj/machinery/computer/power_monitor/ui_interact(mob/user, var/datum/tgui/ui = null)
```
Comment (LetterN) [inline code/modules/power/sensors/sensor_monitoring.dm:53]:
```suggestion
/obj/machinery/computer/power_monitor/ui_interact(mob/user, datum/tgui/ui)
```

File: tgui/packages/tgui/interfaces/Apc.js [hunk 7]
```diff
@@ -194,23 +194,6 @@ const ApcContent = (props, context) => {
                 disabled={locked}
                 onClick={() => act('cover')} />
             )} />
```
Comment (LetterN) [inline tgui/packages/tgui/interfaces/Apc.js:196]: revert the deletion and comment it out

File: code/modules/power/apc.dm (hunk 2 above)
Comment (TheLordME) [reply code/modules/power/apc.dm:770]: attck_hand calles the interact proc, which then calls the ui_interact

**Comment by TheLordME** [comment]
DNM last commit was a mistake

**Comment by TheLordME** [comment]
Fixed the issue, RTM again

**Comment by TheLordME** [comment]
DNM until #2771 is resolved

**Comment by TheLordME** [comment]
issue #2771 
was resolved
this PR doesnt render AIs useless anymore

**Comment by TheLordME** [comment]
No clue, but sounds, fancy chat and tgui are broken now

**Comment by TheLordME** [comment]
fuck this

**Comment by TheLordME** [comment]
still broken, people wanna see my failures

**Pull request closed without merging**